website/docs/proposals/20250630-kaito-cli.md

```diff
@@ -0,0 +1,526 @@
+
+**Pros:**
+
+- Seamless with existing `kubectl` usage
+- Leverages RBAC, kube-context, and namespace scoping
+- Easy to distribute via Krew
+- Inherits kube auth lifecycle (no separate login flow)
+
+**Cons:**
+
+- Bound to kubectl's UX constraints
+- Harder to add interactive features or offline preview
+- Requires `kubectl` binary + plugin install for every user
+
+#### Option B – Standalone CLI
+
+A self-contained binary aimed at platform teams and automation pipelines that need functionality beyond what a kubectl plugin can comfortably deliver.
+
+| **Use Case**                             | **Example**                                                                                                      |
+| ---------------------------------------- | ---------------------------------------------------------------------------------------------------------------- |
+| Platform team managing multiple clusters | `kaitoctl status --all --cluster us-east,us-west`                                                                |
+| CI/CD offline manifest render            | `kaitoctl deploy --n workspace-phi --model phi --dry-run > ws.yaml`                                              |
+| Ops analyst estimating GPU cost          | `kaitoctl cost --workspace gen-answers`                                                                          |
+| LoRA tuning with progress bar            | `kaitoctl tune --name social-sentiment --interactive`                                                            |
+| Distributed inference model deployment   | `kaitoctl deploy --name workspace-mixtral --model mixtral --distributed.enabled=true --distributed.partitions=4` |
+
+**Pros:**
+
+- Full UX control: CLI prompts, spinners, YAML preview, etc.
+- Works without kubectl — just kubeconfig
+- Can support offline use cases and CI manifest staging
+- Supports **multi-cluster** targeting from a single command
+
+**Cons:**
+
+- Slightly larger binary with more dependencies
+- Must manage auth lifecycle (client-go, OIDC, token refresh)
+- Requires separate install path (brew, apt, etc.)
+
+### Decision Drivers
+
+1. **Adoption barrier**: If speed of rollout is top priority, Option A excels
+2. **Feature depth**: For advanced, multi-cluster, or offline workflows, Option B is preferable
+3. **Resourcing**: Building both offers the richest UX but doubles initial engineering effort
+
+## Core Features and Commands
+
+### Workspace Management
+
+#### Deployment Commands
+
+```bash
+# Basic inference deployment
+kubectl kaito deploy --name my-workspace --model phi-3.5-mini-instruct --instance-type Standard_NC6s_v3
+
+# Deploy with specific runtime
+kubectl kaito deploy --name my-workspace --model phi-3-mini --runtime vllm
+kubectl kaito deploy --name my-workspace --model falcon-7b --runtime transformers
+
+# Deploy with adapters
+kubectl kaito deploy --name my-workspace --model phi-3-mini \
+  --adapter my-adapter:v1 --adapter-strength 0.8
```

Comment:
Should we account for adding multiple adapters? 

How about adding adapters to a Workspace that's already running? It would be great to leverage dynamic adapter loading from https://docs.vllm.ai/en/stable/features/lora.html#dynamically-serving-lora-adapters

Reply:
Yes. I think we need to support multi adapter in this cli tool as well. We already supported it in workspace   CRD.